js/rollup.config.mjs

```diff
@@ -0,0 +1,78 @@
+import typescript from '@rollup/plugin-typescript';
+import resolve from '@rollup/plugin-node-resolve';
+import commonjs from '@rollup/plugin-commonjs';
+import dts from 'rollup-plugin-dts';
+import json from '@rollup/plugin-json'; // Import the @rollup/plugin-json to handle JSON files
+
+const generateBundleAndTypeBundles = (file) => {
```

Comment:
The bundle generation function lacks error handling and validation. Consider adding:
1. Input file existence check
2. Format specification for .d.ts files
3. Error handling for failed bundle generation
4. Minification configuration for production builds

Example:
```javascript
const generateBundleAndTypeBundles = (file) => {
  // Validate input file exists
  if (!fs.existsSync(`src/${file}.ts`)) {
    throw new Error(`Source file not found: src/${file}.ts`);
  }

  return [{
    input: `src/${file}.ts`,
    output: [{
      file: `dist/${file}.js`,
      format: 'cjs',
      sourcemap: true,
      compact: true // Enable minification
    }],
    plugins: [/* ... */]
  }, {
    input: `src/${file}.ts`,
    output: {
      file: `dist/${file}.d.ts`,
      format: 'es' // Specify format for type declarations
    },
    plugins: [dts()]
  }]
}
```